File: mobile/lib/core/legal/legal_documents.dart
import 'package:flutter/services.dart';

enum LegalDocumentKind { terms, privacy, biometricConsent }

extension LegalDocumentKindX on LegalDocumentKind {
  String get title {
    switch (this) {
      case LegalDocumentKind.terms:
        return 'Terms & Conditions';
      case LegalDocumentKind.privacy:
        return 'Privacy Policy';
      case LegalDocumentKind.biometricConsent:
        return 'Biometric / Face Consent';
    }
  }

  String get assetPath {
    switch (this) {
      case LegalDocumentKind.terms:
        return '../docs/legal/TERMS-AND-CONDITIONS.md';
      case LegalDocumentKind.privacy:
        return '../docs/legal/PRIVACY-POLICY.md';
      case LegalDocumentKind.biometricConsent:
        return '../docs/legal/BIOMETRIC-CONSENT.md';
    }
  }

  String get route {
    switch (this) {
      case LegalDocumentKind.terms:
        return '/legal/docs/terms';
      case LegalDocumentKind.privacy:
        return '/legal/docs/privacy';
      case LegalDocumentKind.biometricConsent:
        return '/legal/docs/biometric-consent';
    }
  }
}

Future<String> loadLegalDocument(LegalDocumentKind kind) async {
  try {
    return await rootBundle.loadString(kind.assetPath);
  } catch (_) {
    // Flutter asset keys sometimes drop the leading "../".
    final fallback = kind.assetPath.replaceFirst('../', '');
    return rootBundle.loadString(fallback);
  }
}

File: admin-web/src/lib/legalDocuments.ts
import termsMarkdown from "../../../docs/legal/TERMS-AND-CONDITIONS.md?raw";
import privacyMarkdown from "../../../docs/legal/PRIVACY-POLICY.md?raw";
import biometricMarkdown from "../../../docs/legal/BIOMETRIC-CONSENT.md?raw";

export type LegalDocumentSlug = "terms" | "privacy" | "biometric-consent";

export type LegalDocument = {
  slug: LegalDocumentSlug;
  title: string;
  markdown: string;
};

export const LEGAL_DOCUMENTS: Record<LegalDocumentSlug, LegalDocument> = {
  terms: {
    slug: "terms",
    title: "Terms & Conditions",
    markdown: termsMarkdown,
  },
  privacy: {
    slug: "privacy",
    title: "Privacy Policy",
    markdown: privacyMarkdown,
  },
  "biometric-consent": {
    slug: "biometric-consent",
    title: "Biometric / Face Consent",
    markdown: biometricMarkdown,
  },
};

export function legalPlainText(markdown: string) {
  return markdown.replace(/^#+\s*/gm, "").replace(/\*\*/g, "").trim();
}
